features/admin.roles.v2/components/wizard-updated/role-basics.tsx

```diff
@@ -118,6 +119,15 @@ export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicPr
     const { organizationType } = useGetCurrentOrganizationType();
     const isSubOrg: boolean = organizationType === OrganizationType.SUBORGANIZATION;
 
+    // Filter out readonly applications and shared parent applications.
+    const filteredApplicationList: ApplicationListItemInterface[] = useMemo(() => {
+        return applicationList?.applications
+            ?.filter((application: ApplicationListItemInterface) =>
+                !RoleConstants.READONLY_APPLICATIONS_CLIENT_IDS.includes(application?.clientId) &&
+                application?.advancedConfigurations?.fragment === false
```

Comment:
Could you check whether there is a way to send this as filter query in the API request itself so that we wouldn't need frontend filtering altogether?

cc: @sadilchamishka

Reply:
This is a bug fix. And these logics are already available in the codebase. Will not change these logics.